src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,686 @@
+static constexpr winrt::guid DEFAULT_COMMAND_PROMPT_GUID{ 0x0caa0dad, 0x35be, 0x5f56, { 0xa8, 0xff, 0xaf, 0xce, 0xee, 0xaa, 0x61, 0x01 } };
 
 // Function Description:
 // - Extracting the value from an async task (like talking to the app catalog) when we are on the
 //   UI thread causes C++/WinRT to complain quite loudly (and halt execution!)
 //   This templated function extracts the result from a task with chicanery.
 template<typename TTask>
-static auto _extractValueFromTaskWithoutMainThreadAwait(TTask&& task) -> decltype(task.get())
+static auto extractValueFromTaskWithoutMainThreadAwait(TTask&& task) -> decltype(task.get())
 {
-    using TVal = decltype(task.get());
-    std::optional<TVal> finalVal{};
-    std::condition_variable cv;
-    std::mutex mtx;
+    std::optional<decltype(task.get())> finalVal;
+    til::latch latch{ 1 };
 
-    auto waitOnBackground = [&]() -> winrt::fire_and_forget {
+    const auto _ = [&]() -> winrt::fire_and_forget {
         co_await winrt::resume_background();
-        auto v{ co_await task };
+        finalVal.emplace(co_await task);
+        latch.count_down();
+    }();
 
-        std::unique_lock<std::mutex> lock{ mtx };
-        finalVal.emplace(std::move(v));
-        cv.notify_all();
-    };
-
-    std::unique_lock<std::mutex> lock{ mtx };
-    waitOnBackground();
-    cv.wait(lock, [&]() { return finalVal.has_value(); });
-    return *finalVal;
+    latch.wait();
+    return finalVal.value();
 }
 
-static std::tuple<size_t, size_t> _LineAndColumnFromPosition(const std::string_view string, ptrdiff_t position)
+// Concatenates the two given strings and returns them as a path.
+std::filesystem::path buildPath(const std::wstring_view& lhs, const std::wstring_view& rhs)
 {
-    size_t line = 1, column = position + 1;
-    auto lastNL = string.find_last_of('\n', position);
-    if (lastNL != std::string::npos)
-    {
-        column = (position - lastNL);
-        line = std::count(string.cbegin(), string.cbegin() + lastNL + 1, '\n') + 1;
-    }
+    std::wstring buffer;
+    buffer.reserve(lhs.size() + rhs.size());
+    buffer.append(lhs);
+    buffer.append(rhs);
+    return { std::move(buffer) };
+}
 
-    return { line, column };
+// This is a convenience method used by the CascadiaSettings constructor.
+// It runs some basic settings layering without relying on external programs or files.
+// This makes it suitable for most unit tests.
+SettingsLoader SettingsLoader::Default(const std::string_view& userJSON, const std::string_view& inboxJSON)
+{
+    SettingsLoader loader{ userJSON, inboxJSON };
+    loader.LayerInboxOntoUser();
+    loader.FinalizeLayering();
+    return loader;
 }
 
-static void _CatchRethrowSerializationExceptionWithLocationInfo(std::string_view settingsString)
+// The SettingsLoader class is an internal implementation detail of CascadiaSettings.
+// Member methods aren't safe against abuse and you need to ensure to call them in a specific order.

```

Comment:
```suggestion
// Member methods aren't safe against misuse and you need to ensure to call them in a specific order.
```

nit: technically better